kebechet/managers/thoth_provenance/thoth_provenance.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+# Kebechet
+# Copyright(C) 2018, 2019 Kevin Postlethwait
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Consume Thoth Output for Kebechet auto-dependency management"""
+
+import hashlib
+import os
+import logging
+import toml
+import re
+import json
+import typing
+from itertools import chain
+from functools import partial
+import pprint
+
+from thamos import lib
+import git
+
+from kebechet.exception import DependencyManagementError
+from kebechet.exception import InternalError
+from kebechet.exception import PipenvError
+from kebechet.managers.manager import ManagerBase
+from kebechet.source_management import Issue
+from kebechet.source_management import MergeRequest
+from kebechet.utils import cloned_repo
+
+
+_LOGGER = logging.getLogger(__name__)
+
+_BRANCH_NAME = "kebechet_thoth"
+
+# Note: We cannot use pipenv as a library (at least not now - version 2018.05.18) - there is a need to call it
+# as a subprocess as pipenv keeps path to the virtual environment in the global context that is not
+# updated on subsequent calls.
+
+
+class ThothProvenanceManager(ManagerBase):
+    """Manage updates of dependencies."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize update manager."""
+        self._repo = None
+        # We do API calls once for merge requests and we cache them for later use.
+        self._cached_merge_requests = None
+        super().__init__(*args, **kwargs)
+
+    @property
+    def repo(self):
+        """Get repository on which we work on."""
+        return self._repo
+
+    @repo.setter
+    def repo(self, repo: git.Repo):
```

Comment:
I see some of the methods/attributes used multiple times in these managers. Could you refactor them out in a way they share shared code? It will help maintain the codebase (e.g. introduce a base class with shared logic).

Reply:
I just placed them in the manager super class, does that work?